### PR TITLE
api for POST checkout

### LINKED ISSUE #97: User checks out api
[Mindmap](https://github.com/egypt-shops/xshop-docs/blob/develop/detailed_design/user_stories/Ecommerce_story/user_checks_out/user_checks_out.png)
you can check django3 by examples for this endpoint

### PR DESCRIPTION
**Pull request by HMMK99**

### CODE REVIEW

**Comment by HMMK99** [comment]
still for 1 shop only

**Review by mmostafa74 — CHANGES_REQUESTED** [review]
Add serializers tests

**Review by shahwan42 — DISMISSED** [review]

**Comment by shahwan42** [comment]
@mmostafa74 Please review it again and see if everything work as desired

if so, merge and let the mobile team implement it.

File: xshop/orders/tests/test_api/test_serializers.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        serializer = OrderSerializer(data=self.payload)
+        self.assertTrue(serializer.is_valid(raise_exception=True))
+        self.assertFalse(serializer.validated_data.get("paid"))
+
+    def test_order_shop_not_found(self):
+        self.payload["shop"] = 10
+        serializer = OrderSerializer(data=self.payload)
+        self.assertRaisesMessage(
+            ValidationError,
+            "object does not exist",
```
Comment (mmostafa74) [inline xshop/orders/tests/test_api/test_serializers.py]: rename object to Shop

File: xshop/orders/tests/test_api/test_serializers.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            serializer.is_valid,
+            raise_exception=True,
+        )
+
+    def test_order_user_not_found(self):
+        self.payload["user"] = 10
+        serializer = OrderSerializer(data=self.payload)
+        self.assertRaisesMessage(
+            ValidationError,
+            "object does not exist",
```
Comment (mmostafa74) [inline xshop/orders/tests/test_api/test_serializers.py]: rename object to User

**Review by mmostafa74 — CHANGES_REQUESTED** [review]

File: xshop/orders/api/urls.py [hunk 3]
```diff
@@ -11,4 +11,5 @@
         name="order_detail_patch",
     ),
     path("", views.OrderListCreateApi.as_view(), name="order_list_create"),
+    path("checkout/", views.CheckoutApi.as_view(), name="checkout_operations"),
```
Comment (shahwan42) [inline xshop/orders/api/urls.py]: why not just `checkout` for the name, instead of `checkout_operations`?

**Review by shahwan42 — CHANGES_REQUESTED** [review]

**Review by mmostafa74 — APPROVED** [review]

**Pull request merged**